src/test/java/seedu/modsuni/model/module/CodeTest.java

```diff
@@ -11,4 +14,23 @@ public void constructor_null_throwsNullPointerException() {
         Assert.assertThrows(NullPointerException.class, () -> new Code(null));
     }
 
+    @Test
+    public void constructor_invalidCode_throwsIllegalArgumentException() {
+        String invalidCode = "#cs1010";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new Code(invalidCode));
+    }
+
+    @Test
+    public void isValidSalary() {
```

Comment:
I think you named your method wrongly.

Reply:
yes it is indeed wrong. Will change it in next commit